alibi_detect/utils/keops/kernels.py

```diff
@@ -0,0 +1,91 @@
+import numpy as np
+from pykeops.torch import LazyTensor
+import torch
+import torch.nn as nn
+from typing import Callable, Optional
+
+
+def sigma_mean(x: LazyTensor, y: LazyTensor, dist: LazyTensor) -> torch.Tensor:
+    """
+    Bandwidth estimation using the mean heuristic.
+
+    Parameters
+    ----------
+    x
+        LazyTensor of instances with dimension [Nx, 1, features].
+    y
+        LazyTensor of instances with dimension [1, Ny, features].
+    dist
+        LazyTensor with dimensions [Nx, Ny], containing the pairwise distances between `x` and `y`.
+
+    Returns
+    -------
+    The computed bandwidth, `sigma`.
+    """
+    n = x.shape[0]
+    if x.shape == y.shape and (dist.min(axis=1) == 0.).all() and \
+            (torch.arange(n) == dist.argmin(axis=1).cpu().view(-1)).all():
+        n_mean = n * (n - 1)
+    else:
+        n_mean = np.prod(dist.shape)
+    sigma = (.5 * dist.sum(1).sum().unsqueeze(-1) / n_mean) ** .5
+    return sigma
+
+
+class GaussianRBF(nn.Module):
+    def __init__(
+        self,
+        sigma: Optional[torch.Tensor] = None,
+        init_sigma_fn: Callable = sigma_mean,
```

Comment:
Should we be consistent with `torch` and `tensorflow` and set this to be `None` by default? Not sure of the reasons for having done it for `torch` and `tensorflow`, perhaps for serialization? @ascillitoe